code/modules/mob/living/carbon/human/species/station/station_special_abilities_vr.dm

```diff
@@ -80,55 +78,61 @@
-				to_chat(src, "Your body was unable to regenerate, what few living cells remain require additional nutrients to complete the process.")
-				reviving = FALSE // so they can try again when they're given a kickstart
+
+		//Dead when hatching
+		if(stat == DEAD)
+			//Check again for nutriment (necessary?)
+			if(hasnutriment())
+				chimera_hatch()
+				visible_message("<span class='danger'><p><font size=4>The lifeless husk of [src] bursts open, revealing a new, intact copy in the pool of viscera.</font></p></span>") //Bloody hell...
 				return
 
-		else if(stat != DEAD) //If they're alive at the time of regenerating.
+			//Don't have nutriment to hatch! Or you somehow died in between completing your revive and hitting hatch.
+			else
+				to_chat(src, "Your body was unable to regenerate, what few living cells remain require additional nutrients to complete the process.")
+				verbs -= /mob/living/carbon/human/proc/hatch
+				reviving = FALSE //So they can try again when they're given a kickstart
+
+		//Alive when hatching
+		else
 			chimera_hatch()
 			visible_message("<span class='danger'><p><font size=4>The dormant husk of [src] bursts open, revealing a new, intact copy in the pool of viscera.</font></p></span>") //Bloody hell...
-			return
-		else
-			return //Runtime prevention.
 
 /mob/living/carbon/human/proc/chimera_hatch()
-	nutrition -= nutrition/2 //Cut their nutrition in half.
-	var/old_nutrition = nutrition //Since the game is being annoying.
+	verbs -= /mob/living/carbon/human/proc/hatch
 	to_chat(src, "<span class='notice'>Your new body awakens, bursting free from your old skin.</span>")
-	var/T = get_turf(src)
-	new /obj/effect/gibspawner/human/scree(T)
-	var/braindamage = brainloss/2 //If you have 100 brainloss, it gives you 50.
-	does_not_breathe = 0 //start breathing again
-	revive() // I did have special snowflake code, but this is easier.
-	weakened = 2 //Not going to let you get up immediately. 2 ticks before you get up. Overrides the above 10000 weaken.
+
+	//Modify and record values (half nutrition and braindamage)
+	var/old_nutrition = nutrition * 0.5
+	var/braindamage = (brainloss * 0.5) + 10 //A little damage from the process.
+
+	//I did have special snowflake code, but this is easier.
+	revive()
 	mutations.Remove(HUSK)
 	nutrition = old_nutrition
-	brainloss = braindamage //Gives them half their prior brain damage.
-	update_canmove()
+	setBrainLoss(braindamage)
+
+	//Drop everything
 	for(var/obj/item/W in src)
 		drop_from_inventory(W)
-	spawn(3600 SECONDS) //1 hour wait until you can revive.
-		reviving = FALSE
-		to_chat(src, "Your body has recovered from the strenuous effort of rebuilding itself.")
-	verbs -= /mob/living/carbon/human/proc/hatch
-	return
+
+	//Unfreeze some things
+	does_not_breathe = FALSE
+	update_canmove()
+	weakened = 2
+
+	//Visual effects
+	var/T = get_turf(src)
+	new /obj/effect/gibspawner/human/scree(T)
+
+	schedule_callback_in(1 HOUR, VARSET_CALLBACK(src, reviving, FALSE))
```

Comment:
This'll leave them stunlocked for an hour after reviving 'cause of the code on line 64 in station_special_vr. I dunno if there's a better var to use though.

Reply:
Are they incapacitated? Because we can just make them sleep instead. Or stunned. I mean are they supposed to be aware of what's going on around them?

Reply:
Making them sleep until they hatch would also work, as long as they can't be woken up early. I think cadence only used weaken instead of sleep so the chimera player could sort-of-RP being passive and useless (or possibly tasty in the event of someone deciding to omf them while they're there helpless.

HOWEVER - is there any reason why the reviving var can't just have multiple states? Like "reviving = 1" for on cooldown, "reviving = 2" for actively reviving, "reviving = 0" for neither.